compiler-rt/lib/mpk_untrusted/tests/unit/alloc_site_handler_tests.cpp

```diff
@@ -69,9 +72,81 @@ TEST(getAllocSite, ValidBetweenAddress) {
 }
 
 TEST(getAllocSite, PointerArithmeticOverflowWraps) {
-  auto ptr = (rust_ptr)(-1); // 0xffffffffffffffff
+  auto ptr = (rust_ptr)(-1); // max pointer size
```

Comment:
you should probably derive ptr from a uintptr_t, since its you're casting from a signed integer here. you're only getting away w/ this because the compiler is being nice here, but it doesn't have to be. you might also assert that the value of ptr is the value you expect.

you can check the ptr width to determine the max numerical value., or just accept that the test will fail if the pointer size is less than 64-bits. We don't care about 32 bit platforms, but we do want to be sure that we don't succeed because we use a certain version of clang to compile our test code.

Reply:
swapped to deriving pointer from `uintptr_t`.